power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -289,50 +280,63 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     // helper function to update vectors of components
     template <class CompType, cache_type_c CacheType>
     void update_component(std::vector<typename CompType::UpdateType> const& components,
```

Comment:
yes, e.g. `std::vector<...> const&` can be renamed to `std::span<...>` (pass-by-value semantics) or `std::ranges::range<...>` (explicit view; more generic but no random access). i do believe that's out of the scope of this PR, though.

Reply:
Definitely, but I wanted to point it out. In such cases (and many other related to this topic in main model), I do believe that using templates is the way to go. Since the complexity will be there anyway, we might as well just place the requirements (pun inteded) in one place and not have so many similar overloads. But this is just my opinion.

Since we'll probably talk about this later, I will resolve then.